src/components/Insertion/Results/index.tsx

```diff
@@ -1,29 +1,35 @@
-import { collection, getDocs } from 'firebase/firestore';
 import { useEffect, useState } from 'react';
 import { useSearchParams } from 'react-router-dom';
-import { db } from '../../../firebaseConfig';
 import { InsertionData } from '../types';
+import { searchClient } from '@algolia/client-search';
+
+const appID = import.meta.env.ANGOLIA_APP_ID;
+const apiKey = import.meta.env.ANGOLIA_API_KEY;
 
 const Results: React.FC = () => {
   const [searchParams] = useSearchParams();
   const [insertions, setInsertions] = useState<InsertionData[]>([]);
   const query = searchParams.get('query');
 
+  const algoliaClient = searchClient(appID, apiKey);
```

Comment:
stai reinizializzando il `searchClient` ad ogni re-render. di solito questo tipo di istanze si tengono in un file separato e si importano gia inizializzate, simile a come hai fatto per firebase

Reply:
Mi creo un file per la configurazione e importo